packages/dds/merge-tree/src/mergeTree.ts

```diff
@@ -907,13 +926,16 @@ export class MergeTree {
 	 */
 	private slideAckedRemovedSegmentReferences(segments: ISegment[]) {
 		// References are slid in groups to preserve their order.
-		let currentSlideGroup: LocalReferenceCollection[] = [];
+		let currentForwardSlideGroup: LocalReferenceCollection[] = [];
```

Comment:
changes here make me think we probably want a better abstraction; code is getting pretty dense for not very good reasons (this was already true before, the change here just makes it slightly even more unwieldy).

Reply:
AB#5205 addresses my concerns here, am fine with doing follow-up as part of that task